packages/platform-core/src/parameter_path_conversions.ts

```diff
@@ -22,19 +23,35 @@ export class ParameterPathConversions {
    */
   static toParameterFullPath(
     backendId: BackendIdentifier | AppId,
-    secretName: string
+    parameterName: string
   ): string {
     if (typeof backendId === 'object') {
-      return getBackendParameterFullPath(backendId, secretName);
+      return getBackendParameterFullPath(backendId, parameterName);
     }
-    return getSharedParameterFullPath(backendId, secretName);
+    return getSharedParameterFullPath(backendId, parameterName);
+  }
+
+  /**
+   * Generate an SSM path for references to other backend resources
+   */
+  static toResourceReferenceFullPath(
+    backendId: BackendIdentifier,
+    referenceName: string
+  ): string {
+    return `/amplify/${RESOURCE_REFERENCE}/${getBackendIdentifierPathPart(
+      backendId
+    )}/${referenceName}`;
   }
```

Comment:
This is why we added [polling](https://github.com/aws-amplify/amplify-backend/pull/1018/files) to the lambda ssm shim. That poller will pick up changes to ssm values within the polling interval (currently 1 minute).

To your question about adding version here, I don't think we can because the version is only known once the value is uploaded to SSM and the service computes the next version. It can't be known at synth time which is what we need in order to compute the name here.